internal/pkg/aws/handlers/archer_user_agent.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package handlers
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/version"
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+const UserAgentHeader = "Archer-User-Agent"
+
+// ArcherUserAgentHandler returns a http request handler that sets a custom user agent to all aws requests
+func ArcherUserAgentHandler() request.NamedHandler {
+	return request.NamedHandler{
+		Name: "ArcherUserAgent",
+		Fn: func(r *request.Request) {
+			userAgent := r.HTTPRequest.Header.Get(UserAgentHeader)
+			r.HTTPRequest.Header.Set(UserAgentHeader,
+				fmt.Sprintf("aws-archer/%s %s (%s) %s", version.Version, version.Platform, runtime.GOOS, userAgent))
```

Comment:
I have a feeling someone will comment suggesting the user agent should be "aws-ecs-cli-v2..." 

I think that would make a lot of sense. But I am going to preemptively play devil's advocate and note that prefixing the user agent with "aws-archer" might be better because the existing ecs-cli has the user agent "aws-ecs-cli...". Making the new user clearly differentiated from the old one could be wise.

Reply:
haha like I said, I'm not married to the naming so I'll let @kohidave chime in